examples/gno.land/p/demo/timelock/timelock.gno

```diff
@@ -0,0 +1,218 @@
+	}
+
+	if tl.timestamps.Has(id.Binary()) {
+		return errors.New("timelockutil: Schedule: operation already scheduled")
+	}
+
+	tl.timestamps.Set(id.Binary(), uint64(time.Now().Unix())+delay)
+
+	std.Emit(
+		"TimeLockScheduled",
+		"id", id.String(),
+		"delay", strconv.FormatInt(int64(delay), 10),
+	)
+
+	return nil
+}
+
+// Remove operation
+func (tl *TimeLockUtil) Remove(id seqid.ID) {
+	tl.timestamps.Remove(id.Binary())
+
+	std.Emit(
+		"TimeLockRemoved",
+		"id", id.String(),
+	)
+}
+
+// Cancels a planned operation
+func (tl *TimeLockUtil) Cancel(id seqid.ID) error {
+	if !tl.IsPending(id) {
+		errors.New("timelock: Cancel: operation not pending")
+	}
+
+	tl.timestamps.Remove(id.Binary())
+
+	std.Emit(
+		"TimeLockCancelled",
+		"id", id.String(),
+	)
+
+	return nil
+}
+
+// Executes a pending operation
+func (tl *TimeLockUtil) Execute(id seqid.ID) error {
+	if !tl.IsPending(id) {
+		errors.New("timelock: Execute: operation not pending")
+	}
+
+	tl.timestamps.Set(id.Binary(), DoneTimestamp)
+
+	std.Emit(
+		"TimeLockExecuted",
+		"id", id.String(),
+	)
+
+	return nil
+}
+
+// Update the minimum lead time for future operations
+func (tl *TimeLockUtil) UpdateDelay(newDelay uint64) error {
+	if std.PrevRealm().Addr() != tl.accessControl.AdminRole {
+		return errors.New("timelock: UpdateDelay: only admin can update delay")
+	}
+
+	std.Emit(
+		"TimeLockMinDelayChanged",
+		"oldDelay", strconv.FormatInt(int64(tl.minDelay), 10),
+		"newDelay", strconv.FormatInt(int64(newDelay), 10),
+	)
+
+	tl.minDelay = newDelay
+
+	return nil
+}
+
+// Checks if an operation is pending
+func (tl *TimeLockUtil) IsPending(id seqid.ID) bool {
+	state, err := tl.GetOperationState(id)
+	if err != nil {
+		// Handle the error appropriately; for now, we assume the operation is not pending if there's an error
+		ufmt.Errorf("Error retrieving operation state: %v", err)
+		return false
+	}
+
+	return state == Pending
+}
+
+// Checks if an operation is ready
+func (tl *TimeLockUtil) IsReady(id seqid.ID) bool {
+	state, err := tl.GetOperationState(id)
+	if err != nil {
+		// Handle the error appropriately; for now, we assume the operation is not pending if there's an error
+		ufmt.Errorf("Error retrieving operation state: %v", err)
+		return false
+	}
+
+	return state == Ready
+}
+
+// Returns the status of an operation
+func (tl *TimeLockUtil) GetOperationState(id seqid.ID) (OperationState, error) {
+	timestamp, err := tl.GetTimestamp(id)
+	if err != nil {
+		return Unset, err
+	}
+
+	if timestamp == 0 {
+		return Unset, nil
+	}
+	if timestamp == DoneTimestamp {
+		return Done, nil
+	} else if timestamp > time.Now().Unix() {
+		return Pending, nil
+	}
+	return Ready, nil
+}
+
+// Returns the timestamp of an operation
+func (tl *TimeLockUtil) GetTimestamp(id seqid.ID) (int64, error) {
+	value, ok := tl.timestamps.Get(id.Binary())
+
+	if !ok {
+		return 0, nil // Returning 0 as the timestamp and nil for no error
+	}
+
+	switch v := value.(type) {
```

Comment:
When is this even possible, why would `t.timestamps` hold different value types? 
Please standardize this

Reply:
[2307](https://github.com/gnolang/gno/pull/2307/commits/af534a0a7b867841dd161d30d0efccfc45f177ec)